sys/xtimer/xtimer_core.c

```diff
@@ -480,13 +479,18 @@ static void _timer_callback(void)
 
         /* fire timer */
         _shoot(timer);
+
+        /* time update after executing callback */
+        if (!timer->trivial_callback) {
+          now = _xtimer_lltimer_now();
```

Comment:
I don't think so. Please note that the current implementation calls _xtimer_lltimer_now() in _timer_left(xx) function, which calls _xtimer_lltimer_now() every time regardless of trivial or non-trivial callbacks. In contrast, this PR calls _xtimer_lltimer_now() only for non trivial callbacks.